### PR TITLE
New chaplain dual wield nullrod

### PR DESCRIPTION
**Pull request by Moltijoe**
big thanks to @BlueishTsunami for more amazing sprites

sick of me only adding more clothing nullrods?
here have a weapon, sike, it's a belt... sorta

alt-click it while it's in your belt slot to draw the swords
the belt-sheathe doesn't give anti-magic, only the swords that get drawn
dropping, throwing, or otherwise losing the blades will just resheathe them instead
attacks in the exact same way as mantis blades (12 force instead of 20, and no wound bonus or AP)

https://user-images.githubusercontent.com/108117184/234486021-31e320d1-b593-49eb-96dc-242117964cc9.mp4

all the sprites used 
image

:cl:  @BlueishTsunami @Moltijoe 
rscadd: Adds new chaplain dual wield nullrod
imageadd: adds sprites for the dual wield nullrod
/:cl:

### CODE REVIEW

**Comment by ChuChu42** [comment]
MACHINE
TURN BACK NOW

this actually fits really nicely with a chaplain with templar armor and wings
another fancy weapon to my arsenal
i love this tough

**Comment by iamthesilly** [comment]
would be funny if it had some kind of interaction with ultra violence

**Comment by Moltijoe** [comment]
> would be funny if it had some kind of interaction with ultra violence

... maybe

**Comment by Marmio64** [comment]
> > would be funny if it had some kind of interaction with ultra violence
> 
> ... maybe

1 extra damage to them

**Comment by Yarinoi** [comment]
**_I'M GOING TO ULTRAKILL YOU_**

File: code/game/objects/items/holy_weapons.dm [hunk 1]
```diff
@@ -442,6 +442,129 @@
...
+	menutab = MENU_WEAPON
+	additional_desc = "Strap the sheathe to your waist, and these blades will never fail you."
+	var/swords = TRUE
+	var/obj/item/nullrod/handedsword/swordright
+	var/obj/item/nullrod/handedsword/other/swordleft
+
+/obj/item/nullrod/dualsword/AltClick(mob/user)
+	. = ..()
+	if(loc != user)
+		to_chat(user, "you awkwardly struggle to pull the blades out of the sheathe while not wearing it.")
```
Comment (tattax) [inline code/game/objects/items/holy_weapons.dm]:
```suggestion
		to_chat(user, span_notice("You awkwardly struggle to pull the blades out of the sheathe while not wearing it."))
```
```suggestion
		user.balloon_alert(user, "you struggle to pull the blades out of the sheathe..."))
```

File: code/game/objects/items/holy_weapons.dm [hunk 2]
```diff
@@ -442,6 +442,129 @@
...
+		swordleft = new(src)
+		swordleft.force = force
+		swordleft.armour_penetration = armour_penetration
+		swordleft.block_chance = block_chance
+		swordleft.wound_bonus = wound_bonus
+		swordleft.bare_wound_bonus = bare_wound_bonus
+		swordleft.sheath = src
+		user.put_in_l_hand(swordleft)
+
+		to_chat(user, "you draw your righteous blades.")
```
Comment (tattax) [inline code/game/objects/items/holy_weapons.dm]: no. stop.
```suggestion
		to_chat(user, span_notice("You draw your righteous blades."))
```
```suggestion
		user.balloon_alert(user, "you unsheathe \the [src].")
```

File: code/game/objects/items/holy_weapons.dm [hunk 3]
```diff
@@ -442,6 +442,129 @@
...
+	var/obj/item/nullrod/handedsword/other/swordleft
+
+/obj/item/nullrod/dualsword/AltClick(mob/user)
+	. = ..()
+	if(loc != user)
+		to_chat(user, "you awkwardly struggle to pull the blades out of the sheathe while not wearing it.")
+		return
+	if(swords)
+		if(LAZYLEN(user.get_empty_held_indexes()) < 2)
+			to_chat(user, "you need both hands free to draw your swords.")
```
Comment (tattax) [inline code/game/objects/items/holy_weapons.dm]:
```suggestion
			to_chat(user, span_warning("You need both hands free to unsheathe \the [src]!"))
```
```suggestion
			user.balloon_alert(user, "you need both hands to unsheathe \the [src]!")
```

File: code/game/objects/items/holy_weapons.dm [hunk 4]
```diff
@@ -442,6 +442,129 @@
...
+/obj/item/nullrod/dualsword/attackby(obj/item/I, mob/living/user, params)
+	. = ..()
+	if(istype(I, /obj/item/nullrod/handedsword))
+		qdel(I)
+
+		var/otherhand = user.get_inactive_held_item()
+		if(istype(otherhand, /obj/item/nullrod/handedsword))
+			qdel(otherhand)
+
+		to_chat(user, "you sheathe your blades.")
```
Comment (tattax) [inline code/game/objects/items/holy_weapons.dm]:
```suggestion
		to_chat(user, span_notice("You sheathe the [src]."))
```
```suggestion
		user.balloon_alert(user, "you sheathe the [src].")
```

File: code/game/objects/items/holy_weapons.dm [hunk 5]
```diff
@@ -442,6 +442,129 @@
...
+	desc = "\"I'm going to ultrakill you!\" -Righteous Judge"
+	icon = 'icons/obj/weapons/swords.dmi'
+	icon_state = "dualleft"
+	item_state = "dualleft"
+
+/obj/item/nullrod/handedsword/attack(mob/living/M, mob/living/user, secondattack = FALSE)
+	. = ..()
+	var/obj/item/nullrod/handedsword/secondsword = user.get_inactive_held_item()
+	if(istype(secondsword, /obj/item/nullrod/handedsword) && !secondattack)
+		sleep(0.2 SECONDS)
```
Comment (tattax) [inline code/game/objects/items/holy_weapons.dm]: no. ----> addtimer(callback)

File: code/game/objects/items/holy_weapons.dm [hunk 6]
```diff
@@ -442,6 +442,129 @@
...
+
+/obj/item/nullrod/handedsword/dropped(mob/user, silent = TRUE)
+	. = ..()
+	if(QDELETED(src))
+		return
+	var/otherhand = user.get_inactive_held_item()
+	if(istype(otherhand, /obj/item/nullrod/handedsword))
+		qdel(otherhand)
+	if(sheath)
+		to_chat("you sheathe your blades.")
```
Comment (tattax) [inline code/game/objects/items/holy_weapons.dm]: how far have we come
```suggestion
		to_chat(user, span_notice("You sheathe your blades."))
```
```suggestion
		user.balloon_alert(user, "you sheathe \the [src].")
```

File: code/game/objects/items/holy_weapons.dm [hunk 7]
```diff
@@ -442,6 +442,129 @@
...
+		sheath.swords = TRUE
+		sheath.update_icon()
+		playsound(user, 'sound/items/sheath.ogg', 25, TRUE)
+	qdel(src)
```
Comment (tattax) [inline code/game/objects/items/holy_weapons.dm]: use forcemove, we don't want swords to restore integrity everytime they move to the bay + prevents exploits

Comment (tattax) [reply code/game/objects/items/holy_weapons.dm]: or maybe you do but still use something else

**Comment by tattax** [comment]
i dislike the file names being so inconsistent tbh

File: code/game/objects/items/holy_weapons.dm (hunk 5 above)
Comment (Moltijoe) [reply code/game/objects/items/holy_weapons.dm]: but mooomm mantis blade gets to do it

**Review by cuackles — APPROVED** [review]
gud sprites, only issue i have is that each sword only has sprites for the specific hand it uses but for how it works i dont think that would be an issue, also dont see a drop item for the swords but same thing, how do they work on one handed poeple tho ?

**Comment by cuackles** [comment]
> gud sprites, only issue i have is that each sword only has sprites for the specific hand it uses but for how it works i dont think that would be an issue, also dont see a drop item for the swords but same thing, how do they work on one handed poeple tho ?

nevermind there is item sprites my brain is just soft and smooth so the mental image of it just slided down and didnt stick

**Pull request merged**